service/src/main/kotlin/no/nav/su/se/bakover/service/brev/BrevServiceImpl.kt

```diff
@@ -68,11 +72,56 @@ internal class BrevServiceImpl(
         dokumentRepo.lagre(dokument, transactionContext)
     }
 
+    override fun distribuerDokumenter() {
+        dokumentRepo.hentDokumenterForDistribusjon()
+            .map { dokumentdistribusjon ->
+                journalførDokument(dokumentdistribusjon)
+                    .mapLeft {
+                        Distribusjonsresultat.Feil.Journalføring(dokumentdistribusjon.id)
+                    }
+                    .flatMap { journalført ->
+                        distribuerDokument(journalført)
+                            .mapLeft {
+                                Distribusjonsresultat.Feil.Brevdistribusjon(journalført.id)
+                            }
+                            .map { distribuert ->
+                                Distribusjonsresultat.Ok(distribuert.id)
+                            }
+                    }
+            }.ifNotEmpty {
+                val ok = this.ok()
+                val feil = this.feil()
+                if (feil.isEmpty()) {
+                    log.info("Journalførte/distribuerte distribusjonsIDene: $ok")
+                } else {
+                    log.error("Kunne ikke journalføre/distribuere distribusjonsIDene: $feil. Disse gikk ok: $ok")
+                }
+            }
+    }
+
+    private sealed class Distribusjonsresultat {
+        sealed class Feil {
+            data class Journalføring(val id: UUID) : Feil()
+            data class Brevdistribusjon(val id: UUID) : Feil()
+        }
+
+        data class Ok(val id: UUID) : Distribusjonsresultat()
+    }
+
+    private fun List<Either<Distribusjonsresultat.Feil, Distribusjonsresultat.Ok>>.ok() =
+        this.filterIsInstance<Either.Right<Distribusjonsresultat.Ok>>()
+            .map { it.value.id.toString() }
+
+    private fun List<Either<Distribusjonsresultat.Feil, Distribusjonsresultat.Ok>>.feil() =
+        this.filterIsInstance<Either.Left<Distribusjonsresultat.Feil>>()
+            .map { it.value }
+
     /**
      * Henter Person fra PersonService med systembruker.
      * Ment brukt fra async-operasjoner som ikke er knyttet til en bruker med token.
      */
-    override fun journalførDokument(dokumentdistribusjon: Dokumentdistribusjon): Either<KunneIkkeJournalføreDokument, Dokumentdistribusjon> {
+    // Internal for testing.
+    internal fun journalførDokument(dokumentdistribusjon: Dokumentdistribusjon): Either<KunneIkkeJournalføreDokument, Dokumentdistribusjon> {
```

Comment:
Regner med de gikk fra public til internal siden de ikke lenger kalles fra jobben utenfra?

Reply:
Yes! Kan gjøres om hvis de trengs. De ble ikke private, for å slippe å skrive om 30 tester.